catalog/rocks/pgrefs.go

```diff
@@ -0,0 +1,222 @@
+package rocks
+
+import (
+	"context"
+	"database/sql/driver"
+	"errors"
+	"strings"
+
+	"github.com/jackc/pgtype"
+	"github.com/treeverse/lakefs/db"
+)
+
+type PgBranch struct {
+	CommitID     CommitID     `db:"commit_id"`
+	StagingToken StagingToken `db:"staging_token"`
+}
+
+func (ps CommitParents) Value() (driver.Value, error) {
+	if ps == nil {
+		return []string{}, nil
+	}
+	vs := make([]string, len(ps))
+	for i, v := range ps {
+		vs[i] = string(v)
+	}
+	return vs, nil
+}
+
+func (ps *CommitParents) Scan(src interface{}) error {
+	p := pgtype.TextArray{}
+	err := p.Scan(src)
+	if err != nil {
+		return err
+	}
+	for _, v := range p.Elements {
+		*ps = append(*ps, CommitID(v.String))
+	}
+	return nil
+}
+
+type PGRefManager struct {
+	db db.Database
+}
+
+func NewPGRefManager(db db.Database) *PGRefManager {
+	return &PGRefManager{db}
+}
+
+func (m *PGRefManager) GetRepository(ctx context.Context, repositoryID RepositoryID) (*Repository, error) {
+	repository, err := m.db.Transact(func(tx db.Tx) (interface{}, error) {
+		repository := &Repository{}
+		err := tx.Get(repository,
+			`SELECT storage_namespace, creation_date, default_branch FROM kv_repositories WHERE id = $1`,
+			repositoryID)
+		if err != nil {
+			return nil, err
+		}
+		return repository, nil
+	}, db.ReadOnly(), db.WithContext(ctx))
+	if errors.Is(err, db.ErrNotFound) {
+		return nil, ErrNotFound
+	}
+	if err != nil {
+		return nil, err
+	}
+	return repository.(*Repository), nil
+}
+
+func (m *PGRefManager) CreateRepository(ctx context.Context, repositoryID RepositoryID, repository Repository, branch Branch) error {
+	_, err := m.db.Transact(func(tx db.Tx) (interface{}, error) {
+		_, err := tx.Exec(
+			`INSERT INTO kv_repositories (id, storage_namespace, creation_date, default_branch) VALUES ($1, $2, $3, $4)`,
+			repositoryID, repository.StorageNamespace, repository.CreationDate, repository.DefaultBranchID)
+		if err != nil {
+			return nil, err
```

Comment:
This returns a DB error when a repo is created twice with the same name.  We had both ways of doing it in the cataloger, I think we should pick one and stick to it throughout.  Just announce that we've decided this way :-/

Reply:
what would you like to see returned?